Sources/Lumina/Camera/Extensions/CaptureDeviceHandlerExtension.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  CaptureDeviceHandlerExtension.swift
+//  Lumina
+//
+//  Created by David Okun on 11/20/17.
+//  Copyright © 2017 David Okun. All rights reserved.
+//
+
+import Foundation
+import AVFoundation
+
+extension LuminaCamera {
+  func getNewVideoInputDevice() -> AVCaptureDeviceInput? {
+    do {
+      guard let device = getDevice(with: self.position == .front ? AVCaptureDevice.Position.front : AVCaptureDevice.Position.back) else {
+        LuminaLogger.error(message: "could not find valid AVCaptureDevice")
+        return nil
+      }
+      let input = try AVCaptureDeviceInput(device: device)
+      return input
+    } catch {
+      return nil
+    }
+  }
+  
+  func getNewAudioInputDevice() -> AVCaptureDeviceInput? {
+    do {
+      guard let device = AVCaptureDevice.default(for: AVMediaType.audio) else {
+        return nil
+      }
+      let deviceInput = try AVCaptureDeviceInput(device: device)
+      return deviceInput
+    } catch {
+      return nil
+    }
+  }
+  
+  func purgeAudioDevices() {
+    LuminaLogger.notice(message: "purging old audio devices on capture session")
+    for oldInput in self.session.inputs where oldInput == self.audioInput {
+      self.session.removeInput(oldInput)
+    }
+  }
+  
+  func purgeVideoDevices() {
+    LuminaLogger.notice(message: "purging old video devices on capture session")
+    for oldInput in self.session.inputs where oldInput == self.videoInput {
+      self.session.removeInput(oldInput)
+    }
+    for oldOutput in self.session.outputs {
+      if oldOutput == self.videoDataOutput || oldOutput == self.photoOutput || oldOutput == self.metadataOutput || oldOutput == self.videoFileOutput {
+        self.session.removeOutput(oldOutput)
+      }
+      if let dataOutput = oldOutput as? AVCaptureVideoDataOutput {
+        self.session.removeOutput(dataOutput)
+      }
+      if #available(iOS 11.0, *) {
+        if let depthOutput = oldOutput as? AVCaptureDepthDataOutput {
+          self.session.removeOutput(depthOutput)
+        }
+      }
+    }
+  }
+  
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dokun1/Lumina/blob/5e323e0c2296f8f5c5b041c942e35a2720d9c81e/Sources/Lumina/Camera/Extensions/CaptureDeviceHandlerExtension.swift#L64"></span>Lines should not have trailing whitespace.<br />
<code>trailing_whitespace</code> <code>CaptureDeviceHandlerExtension.swift:64</code></td>
    </tr>
  </tbody>
</table>